minio/resource_minio_iam_user.go

```diff
@@ -124,10 +125,18 @@ func minioUpdateUser(ctx context.Context, d *schema.ResourceData, meta interface
 		d.SetId(nn.(string))
 	}
 
-	userStatus := UserStatus{
-		AccessKey: iamUserConfig.MinioIAMName,
-		SecretKey: secretKey,
-		Status:    madmin.AccountEnabled,
+	if d.HasChangeExcept(iamUserConfig.MinioSecret ) {
+		userStatus = UserStatus{
+			AccessKey: iamUserConfig.MinioIAMName,
+			SecretKey: iamUserConfig.MinioSecret,
+			Status:    madmin.AccountEnabled,
+		}
+	}else{
+		userStatus = UserStatus{
+			AccessKey: iamUserConfig.MinioIAMName,
+			SecretKey: secretKey,
+			Status:    madmin.AccountEnabled,
+		}	
```

Comment:
This looks like it would break the secret generation/auto-update functionality (anyways, my understanding of it), except that it didn't seem to work anyhow with the validation I did locally so I really don't know.